tests/main/views/test_process_outcomes.py

```diff
@@ -0,0 +1,918 @@
+            {},
+            # initial_data
+            {
+                "completed_at": None,
+                "awarding_organisation_name": "Lambay Freehold",
+                "award_value": 5432.1,
+                "start_date": datetime.date(2010, 12, 12),
+                "end_date": datetime.date(2011, 12, 12),
+            },
+            # put_values
+            {
+                "completed": True,
+            },
+            # expected_status_code
+            200,
+            # expected_response_data
+            {
+                "processOutcome": AnySupersetOf({
+                    "completed": True,
+                    "award": {
+                        "awardingOrganisationName": "Lambay Freehold",
+                        "awardValue": "5432.10",
+                        "startDate": "2010-12-12",
+                        "endDate": "2011-12-12",
+                    },
+                }),
+            },
+        ),
+        (
+            # other_po_data
+            {},
+            # initial_data
+            {
+                "completed_at": None,
+                "result": "none-suitable",
+            },
+            # put_values
+            {
+                "award": {
+                    "awardingOrganisationName": "Talbot de Malahide",
+                },
+            },
+            # expected_status_code
+            400,
+            # expected_response_data
+            {
+                "error": (
+                    "awarding_organisation_name cannot be set for 'none-suitable' "
+                    + "ProcessOutcomes. Attempted to set value 'Talbot de Malahide'"
+                ),
+            },
+        ),
+        (
+            # other_po_data
+            {
+                "completed_at": None,
+                "result": "cancelled",
+            },
+            # initial_data
+            {
+                "completed_at": datetime.datetime(2010, 3, 3, 3, 3, 3),
+                "result": "none-suitable",
+            },
+            # put_values
+            {
+                "completed": False,
+            },
+            # expected_status_code
+            400,
+            # expected_response_data
+            {
+                "error": "Can't un-complete process outcome",
+            },
+        ),
+        (
+            # other_po_data
+            {
+                "completed_at": datetime.datetime(2010, 3, 3, 3, 3, 3),
+                "result": "cancelled",
+            },
+            # initial_data
+            {
+                "completed_at": None,
+                "result": "none-suitable",
+            },
+            # put_values
+            {
+                "completed": True,
+            },
+            # expected_status_code
+            400,
+            # expected_response_data
+            {
+                "error": AnyStringMatching(
+                    r"(Direct award project|Brief) \d+ already has a complete process outcome: \d+"
```

Comment:
Feel like matching against a specific list of things isn't necessary here. We are more futureproof if we don't check the first part of the error message at all?